File: backend/src/main/java/com/vollailelink/backend/dto/CommandeDTO.java
package com.vollailelink.backend.dto;

import com.vollailelink.backend.model.enums.CommandeStatut;
import com.vollailelink.backend.model.enums.ModePaiement;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommandeDTO {
    private Long id;
    private Long demandeId;
    private Long clientId;
    private String clientNom;
    private String clientPrenom;
    private String clientTelephone;
    private Long produitId;
    private String produitNom;
    /**
     * Offre commandee (le lot de production). C'est elle qui porte le stock :
     * deux offres du meme produit n'ont pas la meme disponibilite.
     */
    private Long offreId;
    private Integer quantite;
    private BigDecimal prixUnitaire;
    private BigDecimal prixUnitaireAchat;
    private BigDecimal remisePourcentage;
    private BigDecimal remiseMontant;
    private BigDecimal totalHt;
    private BigDecimal totalTtc;
    private CommandeStatut statut;
    private ModePaiement modePaiement;
    private OffsetDateTime datePaiement;
    private OffsetDateTime dateLivraisonPrev;
    private OffsetDateTime dateLivraisonReelle;
    private String lieuLivraison;
    private String numeroSuivi;
    private String commentaires;
    private Boolean isActive;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: frontend/projects/admin/src/app/dto/commande.dto.ts
export interface CommandeDTO {
  id?: number;
  demandeId?: number;
  clientId: number;
  clientNom?: string;
  clientPrenom?: string;
  clientTelephone?: string;
  produitId: number;
  produitNom?: string;
  /**
   * Offre commandée : c'est elle qui porte le stock.
   * Deux offres du même produit (producteurs différents) ont des stocks distincts.
   */
  offreId?: number;
  quantite: number;
  prixUnitaire: number;
  prixUnitaireAchat?: number;
  remisePourcentage: number;
  remiseMontant: number;
  /** Le backend sérialise en camelCase : totalHt / totalTtc (et non totalHT). */
  totalHt: number;
  totalTtc: number;
  /** EN_ATTENTE_PAIEMENT, PAYEE, EN_PREPARATION, EXPEDIEE, LIVREE, ANNULEE, EXPIREE */
  statut: string;
  modePaiement?: string;
  datePaiement?: string; // ISO date string
  dateLivraisonPrev: string; // ISO date string
  dateLivraisonReelle?: string; // ISO date string
  lieuLivraison: string;
  numeroSuivi?: string;
  commentaires?: string;
  isActive?: boolean;
  createdAt?: string;
  updatedAt?: string;
}
